src/main/java/org/apache/commons/imaging/palette/MostPopulatedBoxesMedianCut.java

```diff
@@ -52,13 +50,13 @@ public boolean performNextMedianCut(final List<ColorGroup> colorGroups,
             if (ignoreAlpha && colorComponent == ColorComponent.ALPHA) {
                 continue;
             }
-            Collections.sort(colorGroup.colorCounts, new ColorComparer(colorComponent));
+            Collections.sort(colorCounts, new ColorCountComparator(colorComponent));
```

Comment:
This is sorted multiple times for each color component. Later a best color component will be chosen, and a final sort will take place.